### PR TITLE
For NERSC CPU machines (and GCP), use serial kokkos backend when no threads in CIME case

### PR DESCRIPTION
**Pull request by ndkeen**
For NERSC CPU machines (and GCP), use the serial kokkos backend when the CIME build is NOT SMP (ie no threading).

Fixes https://github.com/E3SM-Project/scream/issues/1997

[bfb]

### CODE REVIEW

**Comment by ndkeen** [comment]
Tested on all machines affected.

**Review by ambrad — APPROVED** [review]

**Comment by ambrad** [comment]
Testing on Mappy will be enough. @ndkeen, feel free to apply AT:AUTOMERGE.

**Comment by ndkeen** [comment]
If I don't add automerge, what happens?

**Comment by AaronDonahue** [comment]
Nothing, Automerge is a flag to allow github to automatically merge the PR if it passes all checks.  If you are pretty confident that you want a PR to merge once it has passed tests and review you can go ahead and add the label.  We often hold off on adding the label for cases where we may want PR's to merge in a certain order (although we expect them all to pass).

If a PR fails any of the pre-merge checks it won't Automerge, so there is that layer of safety.

**Comment by ndkeen** [comment]
Nothing? So then how does the PR get merged?

**Comment by AaronDonahue** [comment]
> Nothing? So then how does the PR get merged?

You either add the AUTOMERGE label, or if it isn't passing the pre-merge checks it can be manually merged by someone w/ authority to do that.  

I think in this case you can go ahead and add the Automerge label.  This PR has passed all the tests and been approved.  

In general most PR's get the Automerge label.  But we like to have the option to hold off on AUTOMERGE.  For example, I'm working on a PR that passes the AT and did get approval, but I wanted to do a series of tests not in the autotester so I didn't use the label, to give me time.  When I'm happy with it I'll put the label on it.

**Comment by ndkeen** [comment]
I still must be missing something.  If it passes the tests, has been approved, why does it not get merged?

**Comment by AaronDonahue** [comment]
Having the AUTOMERGE label is an extra layer of security.  It's not unreasonable to expect that a PR may pass the tests and even get reviewed but to have the PR not ready to merge.  A typical example is in the case of multiple reviewers - if there is at least 1 approval then AUTOMERGE will kick in.  But it isn't uncommon for multiple reviewers to be looking at different aspects of the PR and for the author to wait until all reviews are in to add the AUTOMERGE label.

The alternative is to always have to manually merge, which could hold up PR's since only a handful of people have that authority.

We specifically don't want to automate things so that a passed reviewed PR gets merged without any way to stop it.  It is safer to default to a proactive "you have to add a automerge label".  

In standard practice, once you are happy that a PR is ready and has been properly reviewed you can add the AUTOMERGE label.

**Comment by ndkeen** [comment]
In summary, there are only 2 ways for a PR to be merged:
1) manual merge -- only for VIP authorities
2) add AUTOMERGE label

**Comment by bartgol** [comment]
Noel, you should be able to click on the green "Merge pull request" button right now. Everyone with write access to the repo can merge. The button is green if all checks pass. AUTOMERGE simply means that the user @E3SM-Autotester will "click" on merge. If the checks don't pass, the button is red, and only admins can manually merge the PR.

**Comment by ndkeen** [comment]
OK, is the "green button" option a third was to merge a PR? Or is it the same as manual merging?

**Comment by ambrad** [comment]
The shiny green button is the preferred manual approach. Unlike for the E3SM repo, we don't merge to master locally and then push `master` to the remote.

**Comment by bartgol** [comment]
I don't have any preference between "manual merge" and "AUTOMERGE". The only difference is in the name of the author, so doesn't really matter. I guess in theory the manual merge offers the possibility of adding extra info in the merge commit (e.g., BFB or other useful strings).

**Comment by ndkeen** [comment]
Thanks for the explanations.

**Pull request merged**
